src/components/astra-editor/plugins/prompt-dialog.ts

```diff
@@ -0,0 +1,318 @@
+        Helvetica,
+        ui-sans-serif,
+        system-ui,
+        sans-serif,
+        Apple Color Emoji,
+        Segoe UI Emoji,
+        Segoe UI Symbol,
+        Noto Color Emoji;
+    }
+
+    .dark.container {
+      color: white;
+    }
+
+    .wrap {
+      position: relative;
+      border: 1px solid #e5e5e5;
+      border-radius: 5px;
+      width: 100%;
+      max-width: 500px;
+      height: auto;
+      display: flex;
+      flex-direction: column;
+      padding: 5px;
+    }
+
+    .dark .wrap {
+      border: 1px solid #262626;
+    }
+
+    .error {
+      display: flex;
+      padding: 5px 10px;
+      color: red;
+      font-size: 13px;
+    }
+
+    textarea {
+      resize: none;
+      overflow: hidden;
+      border: none;
+      padding: 10px;
+      padding-right: 30px;
+      outline: none;
+      box-sizing: border-box;
+      width: 100%;
+      font-family: inherit;
+      background: inherit;
+      color: inherit;
+    }
+
+    .actions {
+      display: flex;
+      padding: 5px 10px;
+      gap: 5px;
+    }
+
+    button {
+      cursor: pointer;
+      display: flex;
+      border-radius: 4px;
+      border: none;
+      padding: 4px 10px;
+      background: inherit;
+      justify-items: center;
+      align-items: center;
+      gap: 4px;
+      color: #888;
+    }
+
+    .primary {
+      background: #404040;
+      color: white;
+    }
+
+    .primary:hover {
+      background: #171717;
+    }
+
+    .primary:disabled,
+    .primary:disabled:hover {
+      opacity: 0.4;
+    }
+
+    .dark .primary {
+      background: #e5e7eb;
+      color: black;
+    }
+
+    .muted {
+      color: #a3a3a3;
+    }
+
+    .dark .muted {
+      color: #737373;
+    }
+  `
+
+  focus() {
+    this.inputRef.value?.focus()
+  }
+
+  select() {
+    this.inputRef.value?.select()
+  }
+
+  triggerGenerate(text: string) {
+    if (text) {
+      this.dispatchEvent(new CustomEvent('generate', { detail: text }))
+    }
+  }
+
+  triggerAccept() {
+    this.dispatchEvent(new CustomEvent('accept'))
+  }
+
+  triggerReject() {
+    this.dispatchEvent(new CustomEvent('reject'))
+  }
+
+  triggerClose() {
+    this.dispatchEvent(new CustomEvent('close'))
+  }
+
+  protected shouldShowAcceptButton() {
+    if (!this.previousPrompt) return false
+    return this.previousPrompt === this.text
+  }
+
+  protected shouldShowSubmitEditButton() {
+    if (!this.previousPrompt) return false
+    return this.previousPrompt !== this.text
+  }
+
+  protected shouldShowGenerateButton() {
+    return !this.previousPrompt
+  }
+
+  protected firstUpdated(_changedProperties: PropertyValues): void {
+    super.firstUpdated(_changedProperties)
+    const inputRef = this.inputRef.value
+
+    // This prevent main editor to steal this event from input
+    if (inputRef) {
+      inputRef.addEventListener('paste', (e) => {
+        e.stopPropagation()
+      })
+
+      inputRef.addEventListener('copy', (e) => {
+        e.stopPropagation()
+      })
```

Comment:
i think we might need a disconnected callback to remove these listeners? or does the garbage collector resolve it properly? I'm always iffy, if you have a better understanding :D ?